src/ioctls/vcpu.rs

```diff
@@ -1569,7 +1578,30 @@ impl VcpuFd {
                 r => Ok(VcpuExit::Unsupported(r)),
             }
         } else {
-            Err(errno::Error::last())
+            let errno = errno::Error::last();
+            let run = self.kvm_run_ptr.as_mut_ref();
+            // From https://docs.kernel.org/virt/kvm/api.html#kvm-run :
+            //
+            // KVM_EXIT_MEMORY_FAULT is unique among all KVM exit reasons in that it accompanies
+            // a return code of ‘-1’, not ‘0’! errno will always be set to EFAULT or EHWPOISON
+            // when KVM exits with KVM_EXIT_MEMORY_FAULT, userspace should assume kvm_run.exit_reason
+            // is stale/undefined for all other error numbers.
+            if ret == -1
+                && (errno == errno::Error::new(libc::EFAULT)
+                    || errno == errno::Error::new(libc::EHWPOISON)
+                        && run.exit_reason == KVM_EXIT_MEMORY_FAULT)
```

Comment:
Shouldn't `errno ==` checks be in their own parenthesis?
E.g. 
```
if (ret == -1)
   && (errno == ... || errno == )
   && (run.exit_reason == ...)
```
because now second `errno == ` check will be `and`ed with `run.exit_reason` check before `or`d with first `errno == ` check.

Reply:
thanks, fixed